docs/developing.rst

```diff
@@ -0,0 +1,87 @@
+Developer Guidelines
+====================
+
+Basics
```

Comment:
I think all of the "Basics" currently included here would make more sense in the CONTRIBUTING file. 

I thought here we'd focus on e.g. unit vs integration tests, move over the section on releasing, explain the organization of the package, etc ?